modules/templates/templates/template-detail.tsx

```diff
@@ -0,0 +1,327 @@
+          />
+        ) : (
+          <>
+            <span>{record.name}</span>
+          </>
+        ),
+    },
+    {
+      title: t("templates.type"),
+      dataIndex: "type",
+      render: (record: TableDataType) =>
+        editingField === record.key ? (
+          <select
+            value={
+              editedValues[record.key]?.type !== undefined
+                ? editedValues[record.key].type
+                : record.type
+            }
+            onChange={(e) => handleTypeChange(record.key, e.target.value)}
+            className={styles.input}
+          >
+            {typeOptions.map((option) => (
+              <option key={option} value={option}>
+                {t(`templates.type-${option}`)}
+              </option>
+            ))}
+          </select>
+        ) : (
+          <>
+            <span>{record.type}</span>
+          </>
+        ),
+    },
+    {
+      title: t("templates.description"),
+      dataIndex: "description",
+      render: (record: TableDataType) =>
+        editingField === record.key ? (
+          <input
+            type="text"
+            value={
+              editedValues[record.key]?.description !== undefined
+                ? editedValues[record.key].description
+                : record.description
+            }
+            onChange={(e) =>
+              handleInputChange(record.key, "description", e.target.value)
+            }
+            className={`${styles.input} ${!editedValues[record.key]?.description?.trim() ? styles.inputError : ""}`}
+          />
+        ) : (
+          <>
+            <span>{record.description}</span>
+          </>
+        ),
+    },
+    {
+      title: t("templates.action"),
+      dataIndex: "key",
+      render: (record: TableDataType) => (
+        <>
+          {editingField === record.key ? (
+            <button
+              onClick={() => handleSave(record.key)}
+              className={styles.actionButton}
+            >
+              <FaSave style={{ color: "#59DBBC" }} />
+            </button>
+          ) : (
+            <>
+              <button
+                onClick={() => handleEdit(record.key)}
+                className={styles.actionButton}
+              >
+                <FaEdit style={{ color: "#59DBBC" }} />
+              </button>
+              <button
+                onClick={() => handleDeleteField(record.key)}
+                className={styles.actionButton}
+              >
+                <FaTrash style={{ color: "#e53e3e" }} />
+              </button>
+            </>
+          )}
+        </>
+      ),
+    },
+  ];
+
+  const tableData: TableDataType[] = template
+    ? Object.entries(template.fields).map(([key, field]) => ({
+        key,
+        name: key,
+        ...field,
+      }))
+    : [];
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <button
```

Comment:
Why we don't reuse the resources here? Don't use directly <button> or <input>, but our common styled resourced